test/filters/test_convolution.cpp

```diff
@@ -41,3120 +41,371 @@
 #include <pcl/pcl_tests.h>
 #include <pcl/filters/convolution.h>
 #include <pcl/point_types.h>
+#include <cmath>
+#include <array>
+#include <algorithm>
 
 using namespace pcl;
+using namespace pcl::common;
+using namespace pcl::filters;
 using namespace pcl::test;
 
-PointCloud<PointXYZI>::Ptr input (new PointCloud<PointXYZI> ());
-Eigen::ArrayXf filter(7);
+std::array<RGB, 5> colormap {
+  RGB(74, 19, 139),
+  RGB(29, 135, 228),
+  RGB(139, 194, 74),
+  RGB(255, 235, 59),
+  RGB(255, 0, 0),
+};
 
-TEST (Convolution, convolveRows)
+RGB interpolate_color(float lower_bound, float upper_bound, float value)
 {
-  using namespace pcl::common;
-  using namespace pcl::filters;
+  if (value <= lower_bound) return colormap[0];
+  if (value >= upper_bound) return colormap.back();
+  float step_size = (upper_bound - lower_bound) / static_cast<float>(colormap.size() - 1);
+  std::size_t lower_index = static_cast<std::size_t>((value - lower_bound) / step_size);
+  value -= (lower_bound + static_cast<float>(lower_index) * step_size);
```

Comment:
I skipped the normalization and used only normal [linear interpolation](https://en.wikipedia.org/wiki/Linear_interpolation#Linear_interpolation_as_approximation) because I didn't want to make it more complicated. Everybody has probably learned linear interpolation at school, normalizing values is less common.